### ISSUE TITLE
Version 74.0.3729.131 not working properly

### ISSUE THREAD

**Issue by NoAvalon** [rtm516/Bookmark-Manager#7]
I just tried this rebuild branch and it changes the bookmark page but only to a white one which says "New bookmarks page...". I also tried it on Vivaldi because the old version is still working on it and I had the same result. Just a blank page with sentence mentioned above.

Chrome version: 74.0.3729.131

**Comment by NaturalHarmonia** [comment]
> I just tried this rebuild branch and it changes the bookmark page but only to a white one which says "New bookmarks page...". I also tried it on Vivaldi because the old version is still working on it and I had the same result. Just a blank page with sentence mentioned above.
> 
> Chrome version: 74.0.3729.131

Rtm516 knows. He said he was planning on rewriting it.

**Comment by NoAvalon** [comment]
> I have started a rewrite to this extension if you would like to contribute or test it please check out the rebuild branch. https://github.com/rtm516/Bookmark-Manager/tree/rebuild

With this, I thought he meant that his first test branch is out. My bad.

**Comment by NaturalHarmonia** [comment]
Can't add any bookmarks as of now.

**Comment by rtm516** [comment]
Sorry for misleading you guys, I haven't got that far with it yet. I'll let you know when I have a working test release.

**Comment by NoAvalon** [comment]
Don't worry! Btw. when using an extension like this you need to have developer mode enabled, right? What I was wondering is, is there a way to disable the "developer mode" warning in chrome or rather whitelist an extension for it to not show up? 
I got this every time whenever I start chrome.

**Comment by rtm516** [comment]
> Don't worry! Btw. when using an extension like this you need to have developer mode enabled, right? What I was wondering is, is there a way to disable the "developer mode" warning in chrome or rather whitelist an extension for it to not show up?
> I got this every time whenever I start chrome.

Not that I know of, it may be possible by packing it and then forcing chrome to load the packed extension.

**Comment by NoAvalon** [comment]
Packing it seems to be the way to go. The warning changes from "disable developer mode extension" warning to "unsupported extension disabled". If you pack it then it gets a unique ID. The packed extension is completely blocked by default but you can whitelist it through Google Chrome policy templates. After that no more pop-up for the whitelisted extensions when I start my browser.

**Comment by rtm516** [comment]
The version on the [rebuild](https://github.com/rtm516/Bookmark-Manager/tree/rebuild) branch is now functional, I have added a todo list in the readme for it, please check that before making an issue where a feature is missing. As I said before I'm happy for people to make pull requests to add features or fix bugs.

**Comment by NaturalHarmonia** [comment]
> The version on the [rebuild](https://github.com/rtm516/Bookmark-Manager/tree/rebuild) branch is now functional, I have added a todo list in the readme for it, please check that before making an issue where a feature is missing. As I said before I'm happy for people to make pull requests to add features or fix bugs.

Would you be able to add a dark mode/dark background option to the extension?

**Comment by rtm516** [comment]
> > The version on the [rebuild](https://github.com/rtm516/Bookmark-Manager/tree/rebuild) branch is now functional, I have added a todo list in the readme for it, please check that before making an issue where a feature is missing. As I said before I'm happy for people to make pull requests to add features or fix bugs.
> 
> Would you be able to add a dark mode/dark background option to the extension?

Sure, I'll add it to the TODO.

**Comment by NaturalHarmonia** [comment]
> The version on the [rebuild](https://github.com/rtm516/Bookmark-Manager/tree/rebuild) branch is now functional, I have added a todo list in the readme for it, please check that before making an issue where a feature is missing. As I said before I'm happy for people to make pull requests to add features or fix bugs.

What do you mean by functional? When I tried it earlier it kind of froze when opening the manager.

**Comment by rtm516** [comment]
> > The version on the [rebuild](https://github.com/rtm516/Bookmark-Manager/tree/rebuild) branch is now functional, I have added a todo list in the readme for it, please check that before making an issue where a feature is missing. As I said before I'm happy for people to make pull requests to add features or fix bugs.
> 
> What do you mean by functional? When I tried it earlier it kind of froze when opening the manager.

Strange, its working fine for me well as far as storing and browsing bookmarks. Can you see if there are any errors in the extensions page, as well as console for the bookmarks page?

**Comment by NaturalHarmonia** [comment]
Two errors: Unrecognized property 'bookmarks_ui' of manifest key 'chrome_ui_overrides'.

Error handling response: TypeError: Cannot read property 'split' of undefined

I can't view the console when opening the bookmarks manager.

**Comment by rtm516** [comment]
> Unrecognized property 'bookmarks_ui' of manifest key 'chrome_ui_overrides'.

That can be ignored because it does in fact exist, chrome just complains it doesn't.


> Error handling response: TypeError: Cannot read property 'split' of undefined

Is there any more information such as line numbers of this so I can try and get to the bottom of it?

**Comment by NoAvalon** [comment]
For me, it's working just fine. At least the features you've already implemented I guess. The bookmark page is taking like 10-15 seconds to load but I can use them without a problem.

**Comment by NaturalHarmonia** [comment]
> > Unrecognized property 'bookmarks_ui' of manifest key 'chrome_ui_overrides'.
> 
> That can be ignored because it does in fact exist, chrome just complains it doesn't.
> 
> > Error handling response: TypeError: Cannot read property 'split' of undefined
> 
> Is there any more information such as line numbers of this so I can try and get to the bottom of it?

No Console errors and a new error appeared. Here's the line number for the second error from before.

**Error handling response: TypeError: Cannot read property 'split' of undefined**

chrome.bookmarks.search({url: tab.url}, function(results) {

The new error doesn't seem to have a line number.

**Unchecked runtime.lastError: No tab with id: 1629.**

Context
_generated_background_page.html
Stack Trace
_generated_background_page.html:0 (anonymous function)

**Comment by rtm516** [comment]
> **Error handling response: TypeError: Cannot read property 'split' of undefined**
> 
> chrome.bookmarks.search({url: tab.url}, function(results) {

It seems like this error is created from the url not following a valid scheme, because I later split it up
`bookmark.url.split("://")[1].split("#")[0]` just to make it look better and like it used to. Can you try adding `console.log(bookmark.url);` to bookmarks.js between line 17 and 18.


> **Unchecked runtime.lastError: No tab with id: 1629.**

That seems to be something caused by chrome sending the extension an invalid tab, should just keep running.

**Comment by NaturalHarmonia** [comment]
Seems the error disappeared but when I try to open the manager it just loads indefinitely. I guess I'll just have to wait until the final or complete release. Maybe then it'll work.

**Comment by NaturalHarmonia** [comment]
Same error appeared again even though I put console.log(bookmark.url); between line 17 and 18. :/

**Comment by rtm516** [comment]
> Same error appeared again even though I put console.log(bookmark.url); between line 17 and 18. :/

All that will do is put a message in console, can you send me  or atleast the last one before the error?

**Comment by NaturalHarmonia** [comment]
let url = bookmark.url;

This?

**Comment by rtm516** [comment]
> let url = bookmark.url;
> 
> This?

 No, then `console.log(bookmark.url);` will if you can send me the output that would be great so i can work out the issue and get it fixed.

**Comment by NaturalHarmonia** [comment]
> > let url = bookmark.url;
> > This?
> 
> No, then `console.log(bookmark.url);` will if you can send me the output that would be great so i can work out the issue and get it fixed.



How exactly do I get the Output?

**Comment by NaturalHarmonia** [comment]
> > How exactly do I get the Output?
> 
> Open chrome developer tools and go to the console tab while the bookmarks page is open.

Nothing appears in the console tab.

**Comment by rtm516** [comment]
> How exactly do I get the Output?

Open chrome developer tools and go to the console tab while the bookmarks page is open.

**Comment by rtm516** [comment]
> Nothing appears in the console tab.

Add me on discord so we can try and debug this rtm516#2107

**Comment by NaturalHarmonia** [comment]
> > Nothing appears in the console tab.
> 
> Add me on discord so we can try and debug this rtm516#2107

Sent request. deoxysxx#3762
